SConstruct

```diff
@@ -956,6 +957,17 @@ if selected_platform in platform_list:
         env.vs_incs = []
         env.vs_srcs = []
 
+    if env["ninja"]:
+        SetOption("experimental", "ninja")
+        env.Tool("ninja")
+
+        # By setting this we allow the user to run ninja by themselves with all
+        # the flags they need, as apparently automatically running from scons
+        # is way slower. This is also needed as, every time the user stops scons
+        # directly, the `purge_flaky_files` method clears `build.ninja` up,
+        # forcing them to re-invoke `scons`.
```

Comment:
This can be remedied on the function itself with whitelisting & using a relative path as a template:

```py
def purge_flaky_files():
    whitelist = ["ninja.build"]
    for build_failure in GetBuildFailures():
        path = build_failure.node.path
        if os.path.isfile(path) and path not in whitelist:
            os.remove(path)
```